File: core/src/main/kotlin/org/bitlap/core/BitlapContext.kt
package org.bitlap.core

import org.apache.hadoop.conf.Configuration
import org.bitlap.common.BitlapConf
import org.bitlap.common.EventBus
import org.bitlap.core.data.impl.BitlapCatalogImpl
import org.bitlap.core.sql.BitlapSqlPlanner

/**
 * Desc: Context with core components.
 *
 * Mail: chk19940609@gmail.com
 * Created by IceMimosa
 * Date: 2021/5/30
 */
object BitlapContext {

    val bitlapConf = BitlapConf()

    val catalog by lazy {
        BitlapCatalogImpl(bitlapConf, Configuration()).apply {
            start()
        }
    }

    val sqlPlanner by lazy {
        BitlapSqlPlanner(catalog)
    }

    val eventBus by lazy {
        EventBus().apply { start() }
    }
}

File: bitlap-core-s/src/main/scala/org/bitlap/core/BitlapContext.scala
/**
 * Copyright (C) 2023 bitlap.org .
 */
package org.bitlap.core

import org.bitlap.common.{ BitlapConf, EventBus }
import org.bitlap.core.catalog.impl.BitlapCatalogImpl
import org.bitlap.core.sql.BitlapSqlPlanner

import org.apache.hadoop.conf.Configuration

/** Context with core components.
 */
object BitlapContext {

  val bitlapConf: BitlapConf    = BitlapConf()
  val hadoopConf: Configuration = Configuration() // TODO (merge bitlap.hadoop.xxx)

  lazy val eventBus: EventBus = {
    val e = EventBus()
    e.start()
    e
  }

  lazy val catalog: BitlapCatalogImpl = {
    val c = BitlapCatalogImpl(bitlapConf, hadoopConf)
    c.start()
    c
  }

  lazy val sqlPlanner: BitlapSqlPlanner = new BitlapSqlPlanner(catalog)
}
